src/main/java/org/folio/circulation/domain/notice/NoticeDescriptor.java

```diff
@@ -0,0 +1,115 @@
+package org.folio.circulation.domain.notice;
+
+import org.joda.time.Period;
+
+public class NoticeDescriptor {
+
+  private String templateId;
+
+  private NoticeFormat noticeFormat;
+
+  private NoticeEventType noticeEventType;
+
+  private NoticeTiming timing;
+
+  private Period timingPeriod;
+
+  private boolean recurring;
+
+  private Period recurringPeriod;
+
+  public NoticeDescriptor(
+    String templateId, NoticeFormat noticeFormat, NoticeEventType noticeEventType,
+    NoticeTiming timing, Period timingPeriod,
+    boolean recurring, Period recurringPeriod) {
+
+    this.templateId = templateId;
+    this.noticeFormat = noticeFormat;
+    this.noticeEventType = noticeEventType;
+    this.timing = timing;
+    this.timingPeriod = timingPeriod;
+    this.recurring = recurring;
+    this.recurringPeriod = recurringPeriod;
+  }
+
+  public String getTemplateId() {
+    return templateId;
+  }
+
+  public NoticeFormat getNoticeFormat() {
+    return noticeFormat;
+  }
+
+  public NoticeEventType getNoticeEventType() {
+    return noticeEventType;
+  }
+
+  public NoticeTiming getTiming() {
+    return timing;
+  }
+
+  public Period getTimingPeriod() {
+    return timingPeriod;
+  }
+
+  public boolean isRecurring() {
+    return recurring;
+  }
+
+  public Period getRecurringPeriod() {
+    return recurringPeriod;
+  }
+
+
+  public static class Builder {
```

Comment:
Is this builder used by production code or only tests?

Reply:
Production only

Reply:
Personally, I'd prefer if it was split into a separate file and named appropriately. I will leave that decision up to you.